fendermint/app/src/cmd/run.rs

```diff
@@ -23,13 +24,23 @@ cmd! {
   }
 }
 
-async fn create_parent_finality(_settings: &Settings) -> anyhow::Result<InMemoryFinalityProvider> {
-    // let provider = InMemoryFinalityProvider::new(
-    //     settings.parent_finality.clone(),
-    //     unimplemented!(),
-    // );
-    // Ok(provider)
-    todo!()
+fn create_parent_finality(
+    settings: &Settings,
+    query: &ParentFinalityQuery<RocksDb, NamespaceBlockstore, AppStore>,
+) -> anyhow::Result<InMemoryFinalityProvider> {
+    let last_committed_finality = query.get_committed_finality()?;
```

Comment:
The other place where this could be called would be when a chain reorg is detected in the parent. In this case, we have to restart, that way this query needs to be called again. But how to do that, at this point, is not 100% clear yet. I think having the struct might be easier for future PRs.